staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_group.go

```diff
@@ -71,31 +71,25 @@ func (g *genGroup) Imports(c *generator.Context) (imports []string) {
 func (g *genGroup) GenerateType(c *generator.Context, t *types.Type, w io.Writer) error {
 	sw := generator.NewSnippetWriter(w, c, "$", "$")
 
-	apiPath := func(group string) string {
```

Comment:
I had to backtrack to find where group is being set to "core" - `Group.NonEmpty()` says:

```
func (g Group) NonEmpty() string {                                                                                                                                              
    if g == "api" {
        return "core"
    }   
    return string(g)
}
```

So where does "api" come from?  I guess that comes from the filesystem layout of the legacy group.

Do you think it would make sense to move legacy code or somehow "normalize" it so these weird special cases go away?